File: cxx/src/Structural/Proxy/TestProxy.cpp
0


//TestProxy.java - testing the Proxy

class TestProxy {
   public static void main(String[] args) {
       System.out.println("TestProxy: instantiating PotOfTeaProxy");
       PotOfTeaInterface potOfTea = new PotOfTeaProxy();
       System.out.println(" ");
       System.out.println("TestProxy: pouring tea");
       potOfTea.pourTea();
   }
}

File: csharp/src/Structural/Proxy/TestProxy.cs
//TestProxy.java - testing the Proxy


namespace Structural.Proxy {

using tap;


class TestProxy {
   public static void Main() {
		PotOfTeaInterface potOfTea = new PotOfTeaProxy();

		 Tapper tap = new Tapper();
       tap.test("TestProxy: pouring tea", potOfTea.pourTea().Equals("Pouring Tea"));
		tap.done();
   }
}

}
